splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/FunctionIT.java

```diff
@@ -782,5 +782,36 @@ public void testLengthOfCastToVarchar() throws Exception {
             assertEquals(2, rs.getInt(1));
         }
     }
+
+    @Test
+    public void testCastToCharCcsidAscii() throws SQLException {
+        try (ResultSet rs = methodWatcher.executeQuery("select cast('abc' as char(5)), cast('abc' as char(5) ccsid ascii)," +
+                "cast('abc' as varchar(5)), cast('abc' as varchar(5) ccsid ascii)")) {
+            rs.next();
+            assertEquals(rs.getString(1), rs.getString(2));
+            assertEquals(rs.getString(3), rs.getString(4));
+        }
+    }
+
+    @Test
+    public void testCastToCharForSbcsData() throws Exception
+    {
+        methodWatcher.executeUpdate("drop table sbcs if exists");
+        methodWatcher.executeUpdate("create table sbcs(a varchar(30) for bit data, b char(30) for bit data)");
+        methodWatcher.executeUpdate("insert into sbcs values (cast('a' as varchar(30) for bit data), cast('b' as char(30) for bit data))");
+        methodWatcher.executeUpdate("insert into sbcs values (case when 1 = 2 then 'dummy' end, case when 1 = 2 then 'dummy' end)");
```

Comment:
You can do `insert into sbcs values (NULL, NULL)` the case trick is only needed when you want a NULL in the select portion of a query I think.

Reply:
Ah, good to know, thanks!